packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -255,24 +281,67 @@ export const prepareAdvancerKit = (
           ctx: AdvancerVowCtx & { tmpSeat: ZCFSeat },
         ) {
-          const { poolAccount, intermediateRecipient, settlementAddress } =
-            this.state;
-          const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
-          tmpSeat.exit();
-          const amount = harden({
-            denom: usdc.denom,
-            value: advanceAmount.value,
-          });
-          const transferOrSendV =
-            destination.chainId === settlementAddress.chainId
-              ? E(poolAccount).send(destination, amount)
-              : E(poolAccount).transfer(destination, amount, {
-                  forwardOpts: { intermediateRecipient },
-                });
-          return watch(transferOrSendV, this.facets.transferHandler, {
-            destination,
-            advanceAmount,
-            ...detail,
+          return asVow(async () => {
+            const {
+              poolAccount,
+              intermediateRecipientAccount,
+              intermediateRecipientAddress,
+              settlementAddress,
+            } = this.state;
+            const { destination, advanceAmount, tmpSeat, ...detail } = ctx;
+            tmpSeat.exit();
+            const amount = harden({
+              denom: usdc.denom,
+              value: advanceAmount.value,
+            });
+
+            /** @type {AccountId} */
+            const accountId = parseAccountIdArg(destination);
+            const chainId = accountId.reference;
+            const info = await when(chainHub.getChainInfo(chainId));
+
+            let transferOrSendV;
+            if (chainId === settlementAddress.chainId) {
+              // send to recipient on Agoric
+
+              /** @type {CosmosChainAddress} */
+              const cosmosAddress = makeCosmosAccountId(
+                chainId,
+                'bech32',
+                accountId.accountAddress,
+              );
+              transferOrSendV = E(poolAccount).send(cosmosAddress, amount);
+            } else if (info.namespace === 'cosmos') {
+              // send via IBC
+
+              transferOrSendV = E(poolAccount).transfer(destination, amount, {
+                forwardOpts: {
+                  intermediateRecipient: intermediateRecipientAddress,
+                },
+              });
+            } else if (info.cctpDestinationDomain) {
+              // send USDC via CCTP
+              assert(
+                intermediateRecipientAccount,
+                'intermediateRecipientAccount must be set',
+              );
+
+              // assets are on noble, transfer to dest.
+
+              const encoding = 'ethereum'; // XXX could be solana?
+              transferOrSendV = intermediateRecipientAccount.depositForBurn(
+                `${chainInfoCaipId(info)}:${accountId.accountAddress}`,
+                amount,
+              );
+            } else {
+              Fail`can only transfer to Agoric addresses, via IBC, or via CCTP`;
+            }
```

Comment:
We need to transfer to the Noble ICA before calling `.depositForBurn`: `E(poolAccount).transfer(intermediateRecipientAddress, amount)`

Reply:
done